src/IssueViz/IssueVisualizationControl/ViewModels/IssueVisualizationViewModel.cs

```diff
@@ -0,0 +1,207 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.IssueVisualization.IssueVisualizationControl.ViewModels
+{
+    internal sealed class IssueVisualizationViewModel : INotifyPropertyChanged, IDisposable
+    {
+        private readonly IAnalysisIssueSelectionService selectionEvents;
+        private readonly IVsImageService2 vsImageService;
+        private readonly IRuleHelpLinkProvider ruleHelpLinkProvider;
+        private readonly ILogger logger;
+
+        private IAnalysisIssueVisualization currentIssue;
+        private IAnalysisIssueFlowVisualization currentFlow;
+        private LocationListItem currentLocation;
+
+        private bool isBindingUpdatedOutsideOfControl;
```

Comment:
There is a two-way binding between the panel and the selectionService - this is a way to avoid the infinite recursion. 

Value changed in UI --> the view model updates the property in selectionService --> selectionService raises an event of flow/location Changed --> view model listens to it and calls NotifyPropertyChanged

Reply:
That feels pretty hacky, but I guess it works and I don't have any better suggestions.
It would be worth adding that explanation as a comment in the code.